### PR TITLE
Add ERC20 token support

### PR DESCRIPTION
**Pull request by austinchandra**
Add capability to:
- Add/manage ERC-20 tokens
- Fetch user balances
- Show hard-coded ERC-20 currencies

## Details
This PR builds infrastructure for ERC-20 tokens, similar to CW20 tokens, with a few distinctions. ERC20 queries differ from CW20 and Secret queries in that they query a JSON-RPC endpoint, rather than an HTTP endpoint. The infrastructure is built around this difference, while maintaining principles from the CW20 and Secret implementations.

## Architecture
`erc20/contract-query.ts` 
- The base layer of the ERC-20 module—has the ability to query an ERC-20 contract for all relevant fields. The `ObservableQueryERC20ContractData` class parallels the `ObservableCosmwasmContractChainQuery` class for CW20s, and serves as an extendable object to fulfill contract requests. 
- This file is lengthier than its counterparts because it requires additional infrastructure to fetch and parse ERC-20 contract data using the `ObservableJsonRPCQuery` class

`erc20/erc20-balance.ts`
- Connects the ERC-20 module to the extension's `BalanceRegistry` by wrapping `ObservableQueryERC20ContractData`, as a parallel to `cw20-balance.ts`

`erc20/erc20-contract-info.ts`
- Wraps `ObservableQueryERC20ContractData` to provide a `tokenInfo()` response, to match the behavior provided in `cw20-contract-info.ts`
- This class does not fetch balances and is used primarily when adding a new ERC-20 token in the UI

`erc20/queries.ts`
- Wraps `ObservableQueryERC20ContractData` to provide a root query object, similar to `cosmwasm/queries.ts`

`erc20/transaction-client.ts`
- Creates an Ethereum Tx client to interface with a chain's Ethereum JSON-RPC provider
- This class can format Ethereum Tx's, broadcast signed Tx's to the node, and wait for confirmation

`erc20/types.ts`
- Contains type definitions for the module

`accounts/ethereum.ts`
- This class is responsible for sending ERC-20 token transfers when requested by the UI. It uses Keplr's `signEthereum` and the `erc20/transaction-client` to do so.

## Notes
Support has not yet been implemented for mobile.

## Demo
Token Management
https://user-images.githubusercontent.com/11379797/194942357-f0efaefc-16cb-4107-99ab-a5fadf78dacc.mov
Token Transfer via EVM
https://user-images.githubusercontent.com/11379797/208807580-c6fedd5d-e972-443a-bdf4-b682abcb4f92.mov
IBC Transfer of ERC-20 Token
https://user-images.githubusercontent.com/11379797/209244773-860ebca7-4e4b-4e70-82ee-2c083f8d023c.mov

### CODE REVIEW

**Comment by Thunnini** [comment]
@keplr-wallet/store-etc is uploaded in npm, and also used in other projects as well. We cannot merge this PR when @keplr-wallet/store-etc is deleted

**Comment by Thunnini** [comment]
To be honest, showing only balance (without support transfer) to the user has alot of parts that we need to think about. Also, as of right now, this feature is only for evmos and if features being added start becoming domain specific, it becomes harder and harder to manage code. Current code needs to go through refactoring in the near future, so implementing a special feature is a huge burden in terms of code management. Also, I do think that current PR cycle is very slow, and that you might be unhappy about it.
The suggested way is the following: For evmos, instead of showing assets, we can try linking to your dashboard. This way I think we both can implement this faster and come to an stable agreement. I say we try this first and once this becomes stabilized, adding this feature to Keplr. What do you think of?

**Review by Thunnini — CHANGES_REQUESTED** [review]

File: packages/hooks/src/ibc/amount.ts
Comment (austinchandra) [reply packages/hooks/src/ibc/amount.ts]: Refactored to use the `evmos-erc20` feature flag. The purpose for this condition is that Evmos supports sending ERC-20 tokens in IBC transactions, but other ERC-20 chains may not.

Let me know if this is not a concern, and we can remove the check.

File: packages/stores/src/query/erc20/transaction-client.ts
Comment (austinchandra) [reply packages/stores/src/query/erc20/transaction-client.ts:41]: I'm not sure what you mean. Transaction Client's `createERC20TokenTransferTx` uses `createSignableEthereumTx`, and `account/ethereum` uses `createERC20TokenTransferTx`. 

Thus, it doesn't make sense for `account/ethereum` to call `createERC20TokenTransferTx` in transaction client, which calls `createSignableEthereumTx` once again in `account/ethereum`.

**Comment by HeesungB** [comment]
@austinchandra Thank you for contribution!

**Pull request merged**
